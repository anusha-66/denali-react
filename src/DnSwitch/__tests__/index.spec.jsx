/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render, mount } from 'enzyme';
import { DnSwitch } from '../../index'; // src/index.tsx

describe('DnSwitch Tests', () => {
    it('should render with default props', () => {
        expect.assertions(2);

        const wrapper = mount(<DnSwitch />);

        expect(wrapper.render()).toMatchSnapshot('before change');

        wrapper.find('input[type="checkbox"]').simulate('change');

        expect(wrapper.render()).toMatchSnapshot('after change');
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnSwitch
                id="switch1"
                name="canView"
                offLabel="Off"
                onLabel="On"
                checked
                isDisabled
                className="additional--class"
            />,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should call on change correctly', () => {
        expect.assertions(1);

        const mockedOnChange = jest.fn();
        const wrapper = mount(<DnSwitch onChange={mockedOnChange} />);
        wrapper.find('input[type="checkbox"]').simulate('change');

        expect(mockedOnChange).toHaveBeenCalledTimes(1);
    });
});
