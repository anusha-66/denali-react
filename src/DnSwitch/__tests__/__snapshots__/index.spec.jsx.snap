// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnSwitch Tests should render correctly with custom props 1`] = `
<label
  class="switch additional--class"
>
  <input
    checked=""
    disabled=""
    id="switch1"
    name="canView"
    type="checkbox"
  />
  <span
    class="slider"
  />
  <span
    class="off label"
  >
    Off
  </span>
  <span
    class="on label"
  >
    On
  </span>
</label>
`;

exports[`DnSwitch Tests should render with default props: after change 1`] = `
<label
  class="switch"
>
  <input
    type="checkbox"
  />
  <span
    class="slider"
  />
</label>
`;

exports[`DnSwitch Tests should render with default props: before change 1`] = `
<label
  class="switch"
>
  <input
    type="checkbox"
  />
  <span
    class="slider"
  />
</label>
`;
